src/components/vm/VmInitializer.tsx

```diff
@@ -21,161 +8,48 @@ import {
-import React, { useCallback, useEffect, useState } from 'react';
+import React, { useEffect } from 'react';
+import { useDispatch } from 'react-redux';
 
 import { useEthersProviderContext } from '@/data/web3';
-import { setupFastAuth } from '@/lib/selector/setup';
-import { useAuthStore } from '@/stores/auth';
+import { handleSignIn } from '@/redux/slices/account';
+import type { AppDispatch } from '@/redux/store';
 import { useVmStore } from '@/stores/vm';
-import { recordWalletConnect, reset as resetSegment } from '@/utils/analytics';
-import { networkId, signInContractId } from '@/utils/config';
-import { KEYPOM_OPTIONS } from '@/utils/keypom-options';
+import { recordWalletConnect } from '@/utils/analytics';
+import { networkId } from '@/utils/config';
+import { walletSelector } from '@/utils/wallet';
 
 export default function VmInitializer() {
-  const [signedIn, setSignedIn] = useState(false);
-  const [signedAccountId, setSignedAccountId] = useState(null);
-  const [availableStorage, setAvailableStorage] = useState<Big | null>(null);
-  const [walletModal, setWalletModal] = useState<WalletSelectorModal | null>(null);
   const ethersProviderContext = useEthersProviderContext();
-  const router = useRouter();
   const { initNear } = useInitNear();
   const near = useNear();
   const account = useAccount();
   const cache = useCache();
-  const accountId = account.accountId;
-  const setAuthStore = useAuthStore((state) => state.set);
   const setVmStore = useVmStore((store) => store.set);
 
-  useEffect(() => {
-    initNear &&
-      initNear({
-        networkId,
-        walletConnectCallback: recordWalletConnect,
-        selector: setupWalletSelector({
-          network: networkId,
-          modules: [
-            setupNearWallet(),
-            setupMyNearWallet(),
-            setupSender(),
-            setupHereWallet(),
-            setupMeteorWallet(),
-            setupNeth({
-              gas: '300000000000000',
-              bundle: false,
-            }),
-            setupNightly(),
-            setupWelldoneWallet(),
-            setupFastAuth({
-              networkId,
-              signInContractId,
-              relayerUrl:
-                networkId === 'testnet'
-                  ? 'http://34.70.226.83:3030/relay'
-                  : 'https://near-relayer-mainnet.api.pagoda.co/relay',
-            }) as any, // TODO: Refactor setupFastAuth() to TS
-            setupKeypom({
-              trialBaseUrl:
-                networkId == 'testnet' ? 'https://test.near.org/#trial-url/' : 'https://near.org/#trial-url/',
-              networkId,
-              trialSplitDelim: '/',
-              signInContractId,
-              modalOptions: KEYPOM_OPTIONS(networkId),
-            }) as any, // TODO: Refactor setupKeypom() to TS
-          ],
-        }),
-      });
-  }, [initNear]);
+  const dispatch = useDispatch<AppDispatch>();
 
   useEffect(() => {
-    if (!near) {
+    if (!near || !account) {
```

Comment:
That's because we want to dispatch `handleSignIn` only once when we have `near` initialized and we have `account` data. I think that overall, we should refactor `VmInitializer` from dynamic import to regular import, that will allow us to use VM hooks in a normal way but it's impossible now because regular import from `near-social-vm` is throwing an error.